### PR TITLE
Add is_json_scalar Presto Json function

### PR DESCRIPTION
**Pull request by pramodsatya**
Adds Json function is_json_scalar to determine if the Json object represents a scalar.

### CODE REVIEW

File: velox/functions/prestosql/JsonFunctions.h [hunk 1]
```diff
@@ -19,6 +19,21 @@
...
+// false or null)
+template <typename T>
+struct isJsonScalarFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      bool& result,
+      const arg_type<Varchar>& json) {
+    auto parsedJson = folly::parseJson(json);
+    result = parsedJson.isNumber() or parsedJson.isString() or parsedJson.isBool() or parsedJson.isNull();
```
Comment (kagamiori) [inline velox/functions/prestosql/JsonFunctions.h:33]: Let's use `||` instead `or` here to be more consistent with other code in Velox.

File: velox/functions/prestosql/JsonFunctions.h [hunk 2]
```diff
@@ -19,6 +19,21 @@
 
 namespace facebook::velox::functions {
 
+// is_json_scalar(json) -> boolean
```
Comment (kagamiori) [inline velox/functions/prestosql/JsonFunctions.h:22]: Could you update velox/docs/functions/json.rst to include this function in the documentation? Feel free to copy the content from the Presto's: https://prestodb.io/docs/current/functions/json.html?highlight=is_json_scalar#is_json_scalar. Then we can remove the comment here.

**Review by kagamiori — COMMENTED** [review]
Hi Pramod, Thank you for adding this Json function! It looks good to me overall. I just left a few comments.

File: velox/functions/prestosql/JsonFunctions.h (hunk 1 above)
Comment (pramodsatya) [reply velox/functions/prestosql/JsonFunctions.h]: Thanks, changed accordingly

File: velox/functions/prestosql/JsonFunctions.h (hunk 2 above)
Comment (pramodsatya) [reply velox/functions/prestosql/JsonFunctions.h]: Sure, it is added now, thanks!

**Comment by kagamiori** [comment]
Thank you, Pramod. Looks good to me. I'm going to import this PR and will keep you updated if any further change is needed.

File: velox/functions/prestosql/tests/JsonFunctionsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  std::optional<bool> is_json_scalar(std::optional<std::string> json) {
+    return evaluateOnce<bool>("is_json_scalar(c0)", json);
+  }
+};
+
+TEST_F(JsonFunctionsTest, isJsonScalar) {
+  // Scalars.
+  EXPECT_EQ(is_json_scalar(R"(1)"), true);
+  EXPECT_EQ(is_json_scalar(R"(123456)"), true);
+  EXPECT_EQ(is_json_scalar(R"("hello")"), true);
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/JsonFunctionsTest.cpp:35]: Should we also add a long string case ? Please add > 12 characters to test out of line behavior.

File: velox/functions/prestosql/registration/JsonFunctionsRegistration.cpp [hunk 4]
```diff
@@ -15,10 +15,11 @@
  */
 
 #include "velox/functions/Registerer.h"
-#include "velox/functions/prestosql/JsonExtractScalar.h"
+#include "velox/functions/prestosql/JsonFunctions.h"
 
 namespace facebook::velox::functions {
 void registerJsonFunctions() {
+  registerFunction<isJsonScalarFunction, bool, Varchar>({"is_json_scalar"});
```
Comment (aditi-pandit) [inline velox/functions/prestosql/registration/JsonFunctionsRegistration.cpp:22]: Uppercase 'I' here. That is the convention for function structures.

**Review by aditi-pandit — COMMENTED** [review]
Thanks Pramod. This is looking good. Few nits.

File: velox/functions/prestosql/registration/JsonFunctionsRegistration.cpp (hunk 4 above)
Comment (pramodsatya) [reply velox/functions/prestosql/registration/JsonFunctionsRegistration.cpp]: Thanks for sharing, modified accordingly

**Review by aditi-pandit — APPROVED** [review]

**Pull request closed without merging**